storage/byodb/snowflake-secure-data-sharing/index.md

```diff
@@ -0,0 +1,106 @@
+
+### 1. Create a Share
+Use `CREATE SHARE` to set up a share. Initially, the share is an empty container waiting for objects and accounts to be added.
+
+```sql
+CREATE SHARE <SHARE_NAME>;
+```
+
+### 2. Add Objects to the Share by Granting Privileges
+Grant privileges using `GRANT <privilege> ... TO SHARE`. You need to provide:
+* USAGE privilege on the database to be shared. 
+* USAGE privilege on each database schema containing the objects to be shared. 
+* SELECT privilege for specific objects in the shared schema:
+  * Tables
+  * External tables
+  * Secure views
+  * Secure materialized views
+
+Example:
+```sql
+GRANT USAGE ON DATABASE <DATABASE_NAME> TO SHARE <SHARE_NAME>;
+GRANT USAGE ON SCHEMA <DATABASE_NAME>.<SCHEMA_NAME> TO SHARE <SHARE_NAME>;
+GRANT SELECT ON ALL TABLES IN DATABASE <DATABASE_NAME> TO SHARE <SHARE_NAME>;
+GRANT SELECT ON ALL TABLES IN SCHEMA <DATABASE_NAME>.<SCHEMA_NAME> TO SHARE <SHARE_NAME>;
+GRANT SELECT ON TABLE <DATABASE_NAME>.<SCHEMA_NAME>.<TABLE_NAME> TO SHARE <SHARE_NAME>;
+```
+
+To review the object grants, use:
+```sql
+SHOW GRANTS TO SHARE <SHARE_NAME>;
+```
+
+### 3. Add Accounts to the Share
+Add one or more accounts to the share using `ALTER SHARE`. To verify, use `SHOW SHARES`.
+```sql
+ALTER SHARE <SHARE_NAME ADD ACCOUNTS=<ACCOUNT_NAME>;
+```
+```sql
+SHOW SHARES;
+```
+
+The share is now ready for consumption by the specified accounts. For detailed guidance, refer to [Create and configure shares](https://docs.snowflake.com/en/user-guide/data-sharing-provider).
+
+## Consumer Workflow
+
+Refer to the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/data-share-consumers) for more information.
+
+### Role Requirements
+To perform these tasks, use the `ACCOUNTADMIN` role or a role with the `IMPORT SHARE` global privilege. For more details, see [Enable non-ACCOUNTADMIN roles to perform data sharing tasks](https://docs.snowflake.com/en/user-guide/security-access-privileges-shares).
+
+### Limitations of Imported Databases
+Imported databases have the following restrictions:
+* They are read-only. Consumers can query data but cannot modify it or create new objects.
+* Unsupported actions include:
+  * Cloning imported databases, schemas, or tables.
+  * Using Time Travel for imported databases or their objects.
+  * Editing comments for imported databases.
+* They cannot be re-shared with other accounts.
+* They cannot be replicated.
+
+### Viewing Available Shares
+Use the web interface or SQL to view available shares:
+```sql
+SHOW SHARES;
+```
+
+```sql
+DESC SHARE <PRODUCER_ACCOUNT>.<SHARE_NAME>;
+```
+
+### Create a Database from a Share
+To consume shared data, create a database from the share:
+```sql
+CREATE DATABASE <CONSUMER_DATABASE_NAME> FROM SHARE <PRODUCER_ACCOUNT>.<SHARE_NAME>;
+```
+
+### Granting privileges on an imported database
+Currently, Keboola supports only shares with **direct access to database objects**. Shares with roles are not supported.
```

Comment:
`Shares with roles` - to by chtelo link do https://docs.snowflake.com/en/user-guide/data-sharing-policy-protected-data

Reply:
jo, to chtělo, link přidán